test/ng/httpSpec.js

```diff
@@ -2369,12 +2369,36 @@ describe('$http param serializers', function() {
                                                                            //a=b&foo[bar]=barv&foo[baz]=bazv
     });
 
+    it('should serialize objects with function properties by repeating param name with [key] suffix', function() {
+      expect(jqrSer({a: function() { return 'b'; }, foo: {'bar': 'barv', 'baz': 'bazv'}})).toEqual(
+        'a=b&foo%5Bbar%5D=barv&foo%5Bbaz%5D=bazv');
+        //a=b&foo[bar]=barv&foo[baz]=bazv
+    });
+
+    it('should serialize objects with function properties returning an object by repeating param name with [key] suffix', function() {
+      expect(jqrSer({a: function() { return { b: 'c' }; }, foo: {'bar': 'barv', 'baz': 'bazv'}})).toEqual(
+        'a%5Bb%5D=c&foo%5Bbar%5D=barv&foo%5Bbaz%5D=bazv');
+        //a[b]=c&foo[bar]=barv&foo[baz]=bazv
+    });
```

Comment:
These two tests are testing more (independent) things than necessary. They should focus on the `function` part. The rest is irrelevant.